pkg/agent/util/net_windows_test.go

```diff
@@ -0,0 +1,1342 @@
+			}
+			defer func() {
+				runCommand = ps.RunCommand
+				netInterfaceByName = net.InterfaceByName
+			}()
+			gotExists := HostInterfaceExists(tc.testNetInterfaceName)
+			assert.Equal(t, tc.wantExists, gotExists)
+		})
+	}
+}
+
+func TestSetInterfaceMTU(t *testing.T) {
+	runCommand = func(cmd string) (string, error) {
+		return "success", nil
+	}
+	defer func() {
+		runCommand = ps.RunCommand
+	}()
+	err := SetInterfaceMTU("test0", 1)
+	assert.NoError(t, err)
+	runCommandErr := fmt.Errorf("invalid")
+	runCommand = func(cmd string) (string, error) {
+		return "", runCommandErr
+	}
+	err = SetInterfaceMTU("test1", 2)
+	assert.Equal(t, runCommandErr, err)
+}
+
+func TestReplaceNetRoute(t *testing.T) {
+	_, testSubnet, _ := net.ParseCIDR("192.168.2.0/32")
+	testRoute := &Route{
+		LinkIndex:         0,
+		DestinationSubnet: testSubnet,
+		GatewayAddress:    net.ParseIP("192.168.2.0"),
+		RouteMetric:       MetricDefault,
+	}
+	tests := []struct {
+		name       string
+		route      *Route
+		commandOut string
+		commandErr error
+		wantErr    error
+	}{
+		{
+			name:       "Replace Route",
+			route:      testRoute,
+			commandOut: "0 192.168.1.0/24 192.168.1.0 256 nil nil nil",
+			commandErr: nil,
+			wantErr:    nil,
+		},
+		{
+			name:       "Get Route Err",
+			route:      testRoute,
+			commandOut: "err 192.168.1.0/24 192.168.1.0 256 nil nil nil",
+			commandErr: nil,
+			wantErr:    fmt.Errorf("failed to parse the LinkIndex"),
+		},
+		{
+			name:       "Get Route Not Exist",
+			route:      testRoute,
+			commandOut: "",
+			commandErr: nil,
+			wantErr:    nil,
+		},
+		{
+			name:       "New Route Err",
+			route:      testRoute,
+			commandOut: "",
+			commandErr: fmt.Errorf("invalid"),
+			wantErr:    fmt.Errorf("invalid"),
+		},
+		{
+			name:       "Duplicate Route",
+			route:      testRoute,
+			commandOut: "0 192.168.2.0/24 192.168.2.0 256 nil nil nil",
+			commandErr: nil,
+			wantErr:    nil,
+		},
+		{
+			name:       "Remove Route Err",
+			route:      testRoute,
+			commandOut: "0 192.168.1.0/24 192.168.1.0 256 nil nil nil",
+			commandErr: fmt.Errorf("invalid"),
+			wantErr:    fmt.Errorf("invalid"),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runCommand = func(cmd string) (string, error) {
+				return tc.commandOut, tc.commandErr
+			}
+			defer func() {
+				runCommand = ps.RunCommand
+			}()
+			gotErr := ReplaceNetRoute(tc.route)
+			if tc.wantErr == nil {
+				assert.NoError(t, gotErr)
+			} else {
+				assert.ErrorContains(t, gotErr, tc.wantErr.Error())
+			}
+		})
+	}
+}
+
+func TestNewNetNat(t *testing.T) {
+	tests := []struct {
+		name       string
+		netNatName string
+		subnetCIDR *net.IPNet
+		commandOut string
+		commandErr error
+		wantErr    error
+	}{
+		{
+			name:       "New Net Nat",
+			netNatName: "test-nat",
+			subnetCIDR: &net.IPNet{
+				IP:   net.ParseIP("192.168.1.21"),
+				Mask: net.CIDRMask(32, 32),
+			},
```

Comment:
Ditto, if subnetCIDR in all test cases are the same, you may remove the field and just define a variable to use it in the test function directly.